Expose Milli.milli to let Zinc be more efficient

retronym opened the pull request:
It can avoid the allocation of the `Path` => `File` conversion
and also save extra calls to `readAttribute` when we are using
`JavaMillis`.

retronym committed:
Expose Milli.milli to let Zinc be more efficient

It can avoid the allocation of the `Path` => `File` conversion
and also save extra calls to `readAttribute` when we are using
`JavaMillis`.

typesafe-tools commented:
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt-validator/parambuild/?sbt=develop&zinc=develop&io=pull/291/head&librarymanagement=develop&util=develop&website=develop&deploy_to_bintray=false&run_sbt_community=false&verbose_dbuild=false&community_verbose=false">this link.</A>

retronym commented:
To enable a change like: https://github.com/sbt/zinc/compare/develop...retronym:topic/classpath-cache-overhead?expand=1

retronym committed:
Make Retry more efficient

  - Avoid boxing of result by specializing
  - Avoid allocations of Left/Right and lambda with an
    imperative implementation
  - Avoid allocations of list of excluded exceptions by caching
    them at call sites.
